### ISSUE TITLE
Viewport height changes upon clicking the page

### ISSUE THREAD

**Issue by DustinBrett** [vercel/next.js#47839]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
Binaries:
  Node: 18.15.0
  npm: N/A
  Yarn: 1.22.19
  pnpm: N/A
Relevant packages:
  next: 13.2.5-canary.27
  eslint-config-next: 13.2.4
  react: 18.2.0
  react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://github.com/DustinBrett/daedalOS

### To Reproduce

This can be reproduced from the base `npx create-next-app@latest` app as soon as it's built. I think this issue happens in general with Chrome/Android but to be specific I am using a Samsung Galaxy S21 with Chrome 111 in Incognito mode. As soon as the page loads, the Vercel footer is not visible. Upon clicking the page, window resize event is triggered and values such as `window.outerHeight` change.

I have spent hours doing the suggestions to resolve issues which are often caused by using 100vh, such as changing to 100%, trying 100dvh, using fill-available in min-height and many other things to trigger reflow, scroll, resize, etc.

Seeing as I can create this issue the moment I do `create-next-app` I have had to come to the conclusion that it is in some way related to Next.js.

### Describe the Bug

Upon loading the page in Incognito the element that is placed at `bottom: 0` is not visible. As soon as anywhere on the page is clicked the footer appears and after monitoring all window/document heights I see they all change, except `window.screen.height`.

As mentioned in reproduction steps, although this viewport issue is common, the solution does not seem to work in this exact case and this issue only happens in Android/Chrome/Incognito.

### Expected Behavior

Upon loading the page the height of the viewport should not change. Elements placed at `bottom: 0` should be visible at the bottom. The height should not change by 100's of pixels.

### Which browser are you using? (if relevant)

Chrome 111.0.5563.116

### How are you deploying your application? (if relevant)

N/A

**Comment by icyJoseph** [comment]
Hi,

Could you please show some screenshots/video capture of what you mean? 

How do you reproduce this with `npx create-next-app@latest` exactly? - I created one and it loads fine.

**Comment by icyJoseph** [comment]
I took me a while, but I finally managed to trigger it, only on actual device though, but I've seen it now. Will try to investigate more and try to make it happen on my machine.

**Comment by icyJoseph** [comment]
@DustinBrett Before I continue, could you try with this HTML?

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>Bug?</title>

    <style>
      .footer {
        display: flex;
        justify-content: center;
        position: fixed;
        width: 100%;
        align-items: flex-end;
        pointer-events: none;
        inset: auto 0 0;
        padding: 2rem;
        height: 200px;
        z-index: 1;
      }
    </style>
  </head>
  <body>
    <main>
      <div class="footer">Am I a bug?</div>

      <div>
        <h1>Hello World</h1>

        <p>tap anywhere</p>
      </div>
    </main>
  </body>
</html>

```

Just want to make sure we are looking at the same issue.

**Comment by DustinBrett** [comment]
> @DustinBrett Before I continue, could you try with this HTML?
> 
> ```
> <!DOCTYPE html>
> <html lang="en">
>   <head>
>     <meta charset="UTF-8" />
>     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
>     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
> 
>     <title>Bug?</title>
> 
>     <style>
>       .footer {
>         display: flex;
>         justify-content: center;
>         position: fixed;
>         width: 100%;
>         align-items: flex-end;
>         pointer-events: none;
>         inset: auto 0 0;
>         padding: 2rem;
>         height: 200px;
>         z-index: 1;
>       }
>     </style>
>   </head>
>   <body>
>     <main>
>       <div class="footer">Am I a bug?</div>
> 
>       <div>
>         <h1>Hello World</h1>
> 
>         <p>tap anywhere</p>
>       </div>
>     </main>
>   </body>
> </html>
> ```
> 
> Just want to make sure we are looking at the same issue.

Sure I will try this code and see if the same effect occurs. I'll post an update later today as soon as I'm setup.

**Comment by icyJoseph** [comment]
Awesome, with that HTML I experienced that in Chrome I had to tap the document to make the fixed element appear.

**Comment by DustinBrett** [comment]
> Awesome, with that HTML I experienced that in Android Chrome I had to tap the document to make the fixed element appear.

Yes the exact same issue happens with that HTML you've provided when I host it via npm's `ws` and open it on my phone like I do in the original issue. Thanks for narrowing it down. I will look deeper into this code you've used to recreate it in order to try and fix this for myself. Do you think this is something that could be fixed on Next.js also or is this normal and I just need to solve it?

**Comment by icyJoseph** [comment]
I haven't really analyzed the CSS/HTML structure that much. Wanted to see if we were seeing the same issue, by reducing the amount of code first.

I guess this might be related to real 100vh which iirc, includes the URL bar. Though, why does a tap on the screen fix it?

There's trick, even packages for the real 100vh issue, but this one feels odd... I ran out of time today though. Keep us posted if you realize what layout issue is at play here.

**Comment by DustinBrett** [comment]
> I haven't really analyzed the CSS/HTML structure that much. Wanted to see if we were seeing the same issue, by reducing the amount of code first.
> 
> I guess this might be related to real 100vh which iirc, includes the URL bar. Though, why does a tap on the screen fix it? The URL bar doesn't go away either, so that's kinda suspicious...
> 
> There's trick, even packages, for the real 100vh issue, but this one feels odd... I ran out of time today though. Keep us posted if you realize what layout issue is at play here.
> 
> Also, this ain't a Next.js issue so, it might be worth closing it. Though we can keep on investigating what's going on.

It would be good to find a solution before we close this as it could be something for Next.js to add to their base app if this is something that is just missing. As mentioned in my original post I have tried many packages to solve this issue such as 100vh, 100%, reflow and various packages that promise to fix this, and none of it worked.

But with this more basic HTML/CSS example I think it will be easier to track down a solution now.

**Comment by icyJoseph** [comment]
Could you try commenting out the inset property?

**Comment by DustinBrett** [comment]
> Could you try commenting out the inset property?

Even with just `position: fixed;` & `bottom: 0;` the issue occurs. Without those the element is not locked to the bottom and it's hard to tell if the viewport resized in height.

**Comment by DustinBrett** [comment]
So I have got it fixed finally. It does seem like a real problem but it is indeed not isolated to Next.js. The fixes I was trying earlier were not working because the correct height needs to be grabbed almost immediately before Next.js has had time to do anything. My new solution stores `document.documentElement.clientHeight` via a `<Script id="initialHeight" strategy="beforeInteractive">` tag in `_document` then in a `useLayoutEffect` I check if `document.documentElement.clientHeight` has changed and if it has I manually set my `<main>` elements height.

https://github.com/DustinBrett/daedalOS/commit/7667dd1d28c3948e16328b0f3374aadb3f90ca2b
